src/http/request.rs

```diff
@@ -49,7 +52,32 @@ impl<B> Request<B> {
     /// Read the Request body.
     #[inline]
     pub fn body(&self) -> Option<&B> { self.body.as_ref() }
-    
+
+    /// Take the Body of this Request
+    #[inline]
+    pub fn take_body(self) -> Option<B> { self.body }
```

Comment:
So as to mirror `Response`, I was thinking that the methods should perhaps be `body_ref()` and `body()`.

Maybe exactly how the body stream is taken from a Request or Response should be different, but if so, sounds like a separate issue to discuss.